app/models/concerns/course/searchable_concern.rb

```diff
@@ -0,0 +1,18 @@
+module Course::SearchableConcern
+  extend ActiveSupport::Concern
+
+  module ClassMethods
```

Comment:
Actually this is just for taking out the searching logic from course model to make it looks cleaner, so you want me to put it back ?